src/main/java/reposense/util/FileUtil.java

```diff
@@ -38,6 +39,41 @@
     private static final String GITHUB_API_DATE_FORMAT = "yyyy-MM-dd";
     private static final ByteBuffer buffer = ByteBuffer.allocate(1 << 11); // 2KB
 
+    /**
+     * Zips all the relevant files and relative folders
+     * @param relativePaths contains the relevant folders to be zipped.
+     * @param relevantFiles contains the relevant files to be zipped.
+     * @param fileTypes contains the file types to be zipped. Only files which are of the type "fileTypes" will be
+     *                  zipped.
+     */
+    public static void zipRelativeFiles(HashSet<Path> relativePaths, HashSet<Path> relevantFiles,
+                                        Path sourcePath, Path outputPath, String... fileTypes) {
+        try (
+                FileOutputStream fos = new FileOutputStream(outputPath + File.separator + ZIP_FILE);
+                ZipOutputStream zos = new ZipOutputStream(fos)
+        ) {
+            Set<Path> allFiles = getFilePaths(sourcePath, fileTypes);
+            for (Path path : allFiles) {
+                String filePath = sourcePath.relativize(path.toAbsolutePath()).toString();
+                if (relativePaths.contains(path.getParent()) || relevantFiles.contains(path.getFileName())) {
```

Comment:
Why this method needs both `relativePaths` and `relevantFiles`? It seems that you deal with them equally.

Reply:
You may just add `summary.json` into the `relativePaths`.

Reply:
My bad for missing `path.getParent()` in the first statement. But anyway, can you try to combine these two sets into one? Because from my perspective, it is unnecessary.

Reply:
Noted; combined those two sets into one in my next commit.